### DISCUSSION TITLE
Latency certifier limits update and draw rates

### DISCUSSION THREAD

**Discussion by alion02** [ppy/osu#21868] [Q&A]
### Type

Performance

### Bug description

Update is limited to 4000hz, draw is limited to 1000hz. Opening the options menu and changing the resolution removes the limits and restores functionality, but bugs out the framerate counter (turns it red).

### Version

2022.1227.0-lazer

### Logs

[network.log](https://github.com/ppy/osu/files/10311453/network.log)
[runtime.log](https://github.com/ppy/osu/files/10311454/runtime.log)
[performance.log](https://github.com/ppy/osu/files/10311455/performance.log)
[updater.log](https://github.com/ppy/osu/files/10311456/updater.log)

**Comment by peppy** [comment]
I’m not sure what you’re seeing or expecting to see but that sounds about normal.

Please provide a video with commentary or similar, showing what you mean.

**Reply by alion02** [reply]
I'm expecting the certifier to run at unlimited update and draw rates, as per [this wiki page](https://github.com/ppy/osu/wiki/Latency-and-unlimited-frame-rates#why-is-the-latency-certifier-running-at-unlimited-draw-frames). The cap makes it virtually impossible to reach 1000hz+ certification.

### Capped
When I start lazer and enter the latency certifier, I get a 4000hz update rate and 1000hz draw rate limit.
![before](https://user-images.githubusercontent.com/22531550/209838702-9f88130d-8c23-4182-8580-127f075de95a.png)
![before graph](https://user-images.githubusercontent.com/22531550/209838708-edcf8c4c-919e-40ac-a3c3-15fb4da6e25c.png)

### Uncapped
If I go into the options menu while in the latency certifier and change the resolution, the limits are removed ("∞"). For some reason this also makes the simple FPS counter unhappy (it turns red).
![after](https://user-images.githubusercontent.com/22531550/209838734-2184e986-6b52-44ab-9a1b-20dd5f66b9b7.png)
![after graph](https://user-images.githubusercontent.com/22531550/209838744-f0d01157-bd69-4c48-8917-cf0510f0747e.png)
